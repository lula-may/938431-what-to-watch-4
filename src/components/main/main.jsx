import React, {PureComponent} from "react";
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import PropTypes from "prop-types";

import GenresList from "../genres-list/genres-list.jsx";
import MoviesList from "../movies-list/movies-list.jsx";
import ShowMoreButton from "../show-more-button/show-more-button.jsx";
import withActiveItem from "../../hocs/with-active-item/with-active-item.jsx";
import withActiveMovie from "../../hocs/with-active-movie/with-active-movie.jsx";

import history from "../../history.js";
import {movieShape} from "../shapes.js";
import {ActionCreator as StateActionCreator} from "../../reducer/app-state/app-state.js";
import {ActionCreator as DataActionCreator, Operation as DataOperation} from "../../reducer/data/data.js";
import {AuthorizationStatus} from "../../reducer/user/user.js";
import {getGenre, getPromoMovie, getUploadingError, getUploadingState, selectMoviesGenres, selectMoviesByGenre} from "../../reducer/data/selectors.js";
import {getMoviesCount} from "../../reducer/app-state/selectors.js";
import {getAuthorizationStatus, getAvatarUrl} from "../../reducer/user/selectors.js";
import {AppRoute} from "../../const.js";

const GenresListWrapped = withActiveItem(GenresList);
const MoviesListWrapped = withActiveMovie(MoviesList);

class Main extends PureComponent {
  constructor(props) {
    super(props);
    this.handleAddToMyListButtonClick = this.handleAddToMyListButtonClick.bind(this);
  }

  handleAddToMyListButtonClick() {
    const {authorizationStatus, promoMovie, onMyListButtonClick} = this.props;
    return authorizationStatus === AuthorizationStatus.NO_AUTH
      ? history.push(AppRoute.LOGIN)
      : onMyListButtonClick(promoMovie);
  }

  render() {
    const {
      activeGenre,
      authorizationStatus,
      avatar,
      genres,
      hasUploadingError,
      isUploading,
      movies,
      moviesCount,
      onGenreClick,
      onMovieCardClick,
      onShowMoreButtonClick,
      promoMovie,
    } = this.props;

    const {
      bgPoster: bgSrc,
      genre: promoMovieGenre,
      poster: posterSrc,
      releaseYear: promoMovieYear,
      title: promoMovieTitle
    } = promoMovie;

    const hasHiddenMovies = movies.length > moviesCount;
    const {isFavorite} = promoMovie;
    const isAuthorized = authorizationStatus === AuthorizationStatus.AUTH;
    const posterAlt = `${promoMovieTitle} poster`;
    const showedMovies = movies.slice(0, moviesCount);

    return <React.Fragment>
      <section className="movie-card">
        <div className="movie-card__bg">
          <img src={bgSrc} alt={promoMovieTitle} />
        </div>

        <h1 className="visually-hidden">WTW</h1>

        <header className="page-header movie-card__head">
          <div className="logo">
            <a className="logo__link">
              <span className="logo__letter logo__letter--1">W</span>
              <span className="logo__letter logo__letter--2">T</span>
              <span className="logo__letter logo__letter--3">W</span>
            </a>
          </div>

          <div className="user-block">
            <Link to={AppRoute.MY_LIST} className="user-block__link">
              {isAuthorized
                ? <div className="user-block__avatar">
                  <img src={avatar} alt="User avatar" width="63" height="63" />
                </div>
                : <span>Sign in</span>
              }
            </Link>
          </div>
        </header>

        <div className="movie-card__wrap">
          <div className="movie-card__info">
            <div className="movie-card__poster">
              <img src={posterSrc} alt={posterAlt} width="218" height="327" />
            </div>

            <div className="movie-card__desc">
              <h2 className="movie-card__title">{promoMovieTitle}</h2>
              <p className="movie-card__meta">
                <span className="movie-card__genre">{promoMovieGenre}</span>
                <span className="movie-card__year">{promoMovieYear}</span>
              </p>

              <div className="movie-card__buttons">
                <Link to={`/films/${promoMovie.id}/player`} className="btn btn--play movie-card__button">
                  <svg viewBox="0 0 19 19" width="19" height="19">
                    <use xlinkHref="#play-s"></use>
                  </svg>
                  <span>Play</span>
                </Link>
                <button
                  className="btn btn--list movie-card__button"
                  type="button"
                  onClick={this.handleAddToMyListButtonClick}
                  disabled={isUploading}
                  style={hasUploadingError ? {border: `1px solid red`} : {}}
                >
                  {isFavorite
                    ? <svg viewBox="0 0 18 14" width="18" height="14">
                      <use xlinkHref="#in-list"></use>
                    </svg>
                    : <svg viewBox="0 0 19 20" width="19" height="20">
                      <use xlinkHref="#add"></use>
                    </svg>
                  }
                  <span>My list</span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </section>

      <div className="page-content">
        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>

          <GenresListWrapped
            activeItem={activeGenre}
            genres={genres}
            onActiveChange={onGenreClick}
          />

          <MoviesListWrapped
            movies={showedMovies}
            moviesCount={moviesCount}
            onMovieCardClick={onMovieCardClick}
          />

          {hasHiddenMovies && <ShowMoreButton onClick={onShowMoreButtonClick}/>}

        </section>
        <footer className="page-footer">
          <div className="logo">
            <a className="logo__link logo__link--light">
              <span className="logo__letter logo__letter--1">W</span>
              <span className="logo__letter logo__letter--2">T</span>
              <span className="logo__letter logo__letter--3">W</span>
            </a>
          </div>

          <div className="copyright">
            <p>Â© 2019 What to watch Ltd.</p>
          </div>
        </footer>
      </div>
    </React.Fragment>;
  }

}

Main.propTypes = {
  activeGenre: PropTypes.string.isRequired,
  authorizationStatus: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired,
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  hasUploadingError: PropTypes.bool.isRequired,
  isUploading: PropTypes.bool.isRequired,
  movies: PropTypes.arrayOf(
      PropTypes.shape(movieShape)
  ).isRequired,
  moviesCount: PropTypes.number.isRequired,
  onGenreClick: PropTypes.func.isRequired,
  onMovieCardClick: PropTypes.func.isRequired,
  onMyListButtonClick: PropTypes.func.isRequired,
  onShowMoreButtonClick: PropTypes.func.isRequired,
  promoMovie: PropTypes.shape(movieShape).isRequired,
};

const mapStateToProps = (state) => ({
  activeGenre: getGenre(state),
  authorizationStatus: getAuthorizationStatus(state),
  avatar: getAvatarUrl(state),
  genres: selectMoviesGenres(state),
  hasUploadingError: getUploadingError(state),
  isUploading: getUploadingState(state),
  movies: selectMoviesByGenre(state),
  moviesCount: getMoviesCount(state),
  promoMovie: getPromoMovie(state),
});

const mapDispatchToProps = (dispatch) => ({
  onGenreClick(genre) {
    dispatch(DataActionCreator.setGenre(genre));
    dispatch(StateActionCreator.resetCount());
  },

  onMovieCardClick(movie) {
    dispatch(DataActionCreator.setActiveMovie(movie));
    dispatch(DataOperation.loadComments(movie.id));
  },

  onShowMoreButtonClick() {
    dispatch(StateActionCreator.incrementMoviesCount());
  },

  onMyListButtonClick(movie) {
    dispatch(DataOperation.updateFavoriteMovies(movie));
  },
});

export {Main};
export default connect(mapStateToProps, mapDispatchToProps)(Main);
