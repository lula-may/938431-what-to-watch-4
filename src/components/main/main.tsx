import * as React from "react";
import {connect} from "react-redux";
import {Link} from "react-router-dom";

import GenresList from "../genres-list/genres-list";
import MoviesList from "../movies-list/movies-list";
import ShowMoreButton from "../show-more-button/show-more-button";
import withActiveItem from "../../hocs/with-active-item/with-active-item";

import history from "../../history";
import {Movie} from "../../types";
import {ActionCreator as StateActionCreator} from "../../reducer/app-state/app-state";
import {Operation as DataOperation} from "../../reducer/data/data";
import {AuthorizationStatus} from "../../reducer/user/user";
import {getGenre} from "../../reducer/app-state/selectors";
import {getPromoMovie, getUploadingError, getUploadingState, selectMoviesGenres, selectMoviesByGenre} from "../../reducer/data/selectors";
import {getMoviesCount} from "../../reducer/app-state/selectors";
import {getAuthorizationStatus, getAvatarUrl} from "../../reducer/user/selectors";
import {AppRoute} from "../../const";

interface Props {
  activeGenre: string;
  authorizationStatus: string;
  avatar: string;
  genres: string[];
  hasUploadingError: boolean;
  isUploading: boolean;
  movies: Array<Movie>;
  moviesCount: number;
  onGenreClick: (genre: string) => void;
  onMovieCardClick: (movie: Movie) => void;
  onMyListButtonClick: (movie: Movie) => void;
  onShowMoreButtonClick: () => void;
  promoMovie: Movie;
}

const GenresListWrapped = withActiveItem(GenresList);

class Main extends React.PureComponent<Props> {
  props: Props;

  constructor(props) {
    super(props);

    this.handleAddToMyListButtonClick = this.handleAddToMyListButtonClick.bind(this);
  }

  handleAddToMyListButtonClick() {
    const {authorizationStatus, promoMovie, onMyListButtonClick} = this.props;
    return authorizationStatus === AuthorizationStatus.NO_AUTH
      ? history.push(AppRoute.LOGIN)
      : onMyListButtonClick(promoMovie);
  }

  render() {
    const {
      activeGenre,
      authorizationStatus,
      avatar,
      genres,
      hasUploadingError,
      isUploading,
      movies,
      moviesCount,
      onGenreClick,
      onMovieCardClick,
      onShowMoreButtonClick,
      promoMovie,
    } = this.props;

    const {
      bgPoster: bgSrc,
      genre: promoMovieGenre,
      poster: posterSrc,
      releaseYear: promoMovieYear,
      title: promoMovieTitle
    } = promoMovie;

    const hasHiddenMovies = movies.length > moviesCount;
    const {isFavorite} = promoMovie;
    const isAuthorized = authorizationStatus === AuthorizationStatus.AUTH;
    const posterAlt = `${promoMovieTitle} poster`;
    const showedMovies = movies.slice(0, moviesCount);

    return <React.Fragment>
      <section className="movie-card">
        <div className="movie-card__bg">
          <img src={bgSrc} alt={promoMovieTitle} />
        </div>

        <h1 className="visually-hidden">WTW</h1>

        <header className="page-header movie-card__head">
          <div className="logo">
            <a className="logo__link">
              <span className="logo__letter logo__letter--1">W</span>
              <span className="logo__letter logo__letter--2">T</span>
              <span className="logo__letter logo__letter--3">W</span>
            </a>
          </div>

          <div className="user-block">
            <Link to={AppRoute.MY_LIST} className="user-block__link">
              {isAuthorized
                ? <div className="user-block__avatar">
                  <img src={avatar} alt="User avatar" width="63" height="63" />
                </div>
                : <span>Sign in</span>
              }
            </Link>
          </div>
        </header>

        <div className="movie-card__wrap">
          <div className="movie-card__info">
            <div className="movie-card__poster">
              <img src={posterSrc} alt={posterAlt} width="218" height="327" />
            </div>

            <div className="movie-card__desc">
              <h2 className="movie-card__title">{promoMovieTitle}</h2>
              <p className="movie-card__meta">
                <span className="movie-card__genre">{promoMovieGenre}</span>
                <span className="movie-card__year">{promoMovieYear}</span>
              </p>

              <div className="movie-card__buttons">
                <Link to={`${AppRoute.FILMS}/${promoMovie.id}${AppRoute.PLAYER}`} className="btn btn--play movie-card__button">
                  <svg viewBox="0 0 19 19" width="19" height="19">
                    <use xlinkHref="#play-s"></use>
                  </svg>
                  <span>Play</span>
                </Link>
                <button
                  className="btn btn--list movie-card__button"
                  type="button"
                  onClick={this.handleAddToMyListButtonClick}
                  disabled={isUploading}
                  style={hasUploadingError ? {border: `1px solid red`} : {}}
                >
                  {isFavorite
                    ? <svg viewBox="0 0 18 14" width="18" height="14">
                      <use xlinkHref="#in-list"></use>
                    </svg>
                    : <svg viewBox="0 0 19 20" width="19" height="20">
                      <use xlinkHref="#add"></use>
                    </svg>
                  }
                  <span>My list</span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </section>

      <div className="page-content">
        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>

          <GenresListWrapped
            activeItem={activeGenre}
            genres={genres}
            onActiveChange={onGenreClick}
          />

          <MoviesList
            movies={showedMovies}
            moviesCount={moviesCount}
            onMovieCardClick={onMovieCardClick}
          />

          {hasHiddenMovies && <ShowMoreButton onClick={onShowMoreButtonClick}/>}

        </section>
        <footer className="page-footer">
          <div className="logo">
            <a className="logo__link logo__link--light">
              <span className="logo__letter logo__letter--1">W</span>
              <span className="logo__letter logo__letter--2">T</span>
              <span className="logo__letter logo__letter--3">W</span>
            </a>
          </div>

          <div className="copyright">
            <p>Â© 2019 What to watch Ltd.</p>
          </div>
        </footer>
      </div>
    </React.Fragment>;
  }
}

const mapStateToProps = (state) => ({
  activeGenre: getGenre(state),
  authorizationStatus: getAuthorizationStatus(state),
  avatar: getAvatarUrl(state),
  genres: selectMoviesGenres(state),
  hasUploadingError: getUploadingError(state),
  isUploading: getUploadingState(state),
  movies: selectMoviesByGenre(state),
  moviesCount: getMoviesCount(state),
  promoMovie: getPromoMovie(state),
});

const mapDispatchToProps = (dispatch) => ({
  onGenreClick(genre) {
    dispatch(StateActionCreator.setGenre(genre));
    dispatch(StateActionCreator.resetCount());
  },

  onMovieCardClick(movie) {
    dispatch(StateActionCreator.setActiveMovie(movie));
    dispatch(DataOperation.loadComments(movie.id));
  },

  onShowMoreButtonClick() {
    dispatch(StateActionCreator.incrementMoviesCount());
  },

  onMyListButtonClick(movie) {
    dispatch(DataOperation.updateFavoriteMovies(movie));
  },
});

export {Main};
export default connect(mapStateToProps, mapDispatchToProps)(Main);
