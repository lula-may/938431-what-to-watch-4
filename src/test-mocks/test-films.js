const movieTitles = [
  `Gone With The Wind`,
  `The Godfather`,
  `Casablanca`,
  `A Streetcar Named Desire`,
  `Some Likes It Hot`,
];

const movies = movieTitles.map((title, i) => {
  return {
    title,
    id: i,
    src: `poster-${i}.jpg`
  };
});

const headerMovie = {
  title: `Jurassic Park`,
  genre: `Sci-Fi`,
  releaseYear: 1993,
  bg: `background.jpg`,
  poster: `header-poster.jpg`
};

export {movies, headerMovie};

