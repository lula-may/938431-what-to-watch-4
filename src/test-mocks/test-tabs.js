const tabs = [
  {
    id: `aa`,
    type: `First`
  },
  {
    id: `bb`,
    type: `Second`,
  },
  {
    id: `cc`,
    type: `Third`,
  }
];

export {tabs};
