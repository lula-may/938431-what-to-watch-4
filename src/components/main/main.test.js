import React from "react";
import renderer from "react-test-renderer";

import {Main} from "./main.jsx";
import {testMovies} from "../../test-mocks/test-films";

const promoMovie = testMovies[0];
const genres = [`All genres`, `Sci-Fi`, `War`, `Detective`, `Fantasy`, `Sport`, `Adventure`];

describe(`Main Component`, () => {
  it(`Should render correctly MainComponent`, () => {
    const tree = renderer.create(
        <Main
          activeGenre={`All genres`}
          authorizationStatus={`AUTH`}
          avatar={`img/avatar.jpg`}
          genres={genres}
          promoMovie={promoMovie}
          movies={testMovies}
          moviesCount={5}
          onGenreClick={() => {}}
          onMovieCardClick={() => {}}
          onPlayButtonClick={() => {}}
          onShowMoreButtonClick={() => {}}
        />
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
