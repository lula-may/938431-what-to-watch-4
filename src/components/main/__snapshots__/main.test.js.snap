// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component Should render correctly MainComponent 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Jurassic Park"
        src="bip-poster-0.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/mylist"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="img/avatar.jpg"
              width="63"
            />
          </div>
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Jurassic Park poster"
            height="327"
            src="poster-0.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Jurassic Park
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Sci-Fi
            </span>
            <span
              className="movie-card__year"
            >
              2000
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/films/0/player"
              onClick={[Function]}
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              disabled={false}
              onClick={[Function]}
              style={Object {}}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            id="All genres"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            id="Sci-Fi"
            onClick={[Function]}
          >
            Sci-Fi
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            id="War"
            onClick={[Function]}
          >
            War
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            id="Detective"
            onClick={[Function]}
          >
            Detective
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            id="Fantasy"
            onClick={[Function]}
          >
            Fantasy
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            id="Sport"
            onClick={[Function]}
          >
            Sport
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            id="Adventure"
            onClick={[Function]}
          >
            Adventure
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/0"
            onClick={[Function]}
          >
            <video
              height="175"
              poster="poster-0.jpg"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/0"
              onClick={[Function]}
            >
              Jurassic Park
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/1"
            onClick={[Function]}
          >
            <video
              height="175"
              poster="poster-1.jpg"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Gone With The Wind
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/2"
            onClick={[Function]}
          >
            <video
              height="175"
              poster="poster-2.jpg"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              The Godfather
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/3"
            onClick={[Function]}
          >
            <video
              height="175"
              poster="poster-3.jpg"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Casablanca
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/4"
            onClick={[Function]}
          >
            <video
              height="175"
              poster="poster-4.jpg"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              A Streetcar Named Desire
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          onClick={[Function]}
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
